pkg/controller/pod/pod_controller.go

```diff
@@ -221,53 +220,40 @@ func (r *ReconcilePod) Reconcile(request reconcile.Request) (reconcile.Result, e
 }
 
 func (r *ReconcilePod) recreateNsxNcpResourceIfDeleted(resName string) error {
-	instance := identifyAndGetInstance(resName)
-	instanceDetails := types.NamespacedName{
-		Namespace: operatortypes.NsxNamespace,
-		Name:      resName,
-	}
-
-	doesResExist, err := r.checkIfK8sResourceExists(instance, instanceDetails)
+	doesResExist, err := operatortypes.CheckIfNCPK8sResourceExists(
+		r.client, resName)
 	if err != nil {
 		log.Error(err, fmt.Sprintf(
-			"Could not retrieve K8s resource - '%s'", instanceDetails.Name))
+			"Could not retrieve K8s resource - '%s'", resName))
```

Comment:
another nit: this error message could be intented as "resource missing". I think we should rephrase to something like "an error occurred while retrieving K8s resource %s", resName

Reply:
done